src/test/java/uk/gov/ons/ctp/common/rest/RestClientTest.java

```diff
@@ -224,40 +203,28 @@ public void testGetResourcesOk() {
   }
 
   /** A test */
-  @Test(expected = ResponseStatusException.class)
-  public void testGetResourcesNoContent() {
-    RestClient restClient = new RestClient();
-    RestTemplate restTemplate = restClient.getRestTemplate();
-
-    MockRestServiceServer mockServer = MockRestServiceServer.createServer(restTemplate);
-    mockServer
-        .expect(requestTo("http://localhost:8080/hotels"))
-        .andExpect(method(HttpMethod.GET))
-        .andRespond(withStatus(HttpStatus.NO_CONTENT));
+  @Test
+  public void testGetResourceNotFound() {
+    mockRequest(HttpStatus.NOT_FOUND, HttpStatus.NOT_FOUND, true);
+  }
 
-    restClient.getResources("/hotels", FakeDTO[].class);
+  /** A test */
+  public void testGetResourcesNoContent() {
+    mockRequest(HttpStatus.NO_CONTENT, HttpStatus.INTERNAL_SERVER_ERROR, false);
   }
 
   /** A test */
-  @Test(expected = ResponseStatusException.class)
   public void testGetResourcesReallyNotOk() {
-    RestClientConfig config =
-        RestClientConfig.builder().scheme("http").host("localhost").port("8080").build();
-    RestClient restClient = new RestClient(config);
-    RestTemplate restTemplate = restClient.getRestTemplate();
-
-    MockRestServiceServer mockServer = MockRestServiceServer.createServer(restTemplate);
-    mockServer
-        .expect(requestTo("http://localhost:8080/hotels"))
-        .andExpect(method(HttpMethod.GET))
-        .andRespond(withStatus(HttpStatus.BAD_REQUEST));
-
-    restClient.getResources("/hotels", FakeDTO[].class);
+    mockRequest(HttpStatus.BAD_REQUEST, HttpStatus.INTERNAL_SERVER_ERROR, true);
   }
 
   /** A test */
-  @Test(expected = ResponseStatusException.class)
+  @Test
   public void testGetResourcesUnauthorized() {
+    mockRequest(HttpStatus.UNAUTHORIZED, HttpStatus.INTERNAL_SERVER_ERROR, true);
+  }
+
+  private void mockRequest(HttpStatus responseStatus, HttpStatus mapStatus, boolean cause) {
```

Comment:
I think this method does a bit more than mock something, so is a bit of a misnomer, but not going to prevent me approving the PR.